specs/core/0_beacon-chain.md

```diff
@@ -1858,16 +1859,19 @@ def process_crosslinks(state: BeaconState) -> None:
     current_epoch = get_current_epoch(state)
     previous_epoch = max(current_epoch - 1, GENESIS_EPOCH)
     next_epoch = current_epoch + 1
+
     for slot in range(get_epoch_start_slot(previous_epoch), get_epoch_start_slot(next_epoch)):
         for crosslink_committee, shard in get_crosslink_committees_at_slot(state, slot):
-            winning_root, participants = get_winning_root_and_participants(state, shard)
+            winning_root, participants = get_winning_root_and_participants(state, slot, shard)
             participating_balance = get_total_balance(state, participants)
             total_balance = get_total_balance(state, crosslink_committee)
             if 3 * participating_balance >= 2 * total_balance:
```

Comment:
I'm not sure I understand the purpose of `previous_crosslinks`. Is the idea that we increase the number scenarios where a crosslink is added? If so, do we want to run the `process_crosslinks` logic over both `current_crosslinks` and `previous_crosslinks`?

Reply:
`prevous_crosslinks` are needed in order to fix this issue https://github.com/ethereum/eth2.0-specs/issues/845. There is a case when attestation created _before_ crosslinks processing is validated _after_ the processing has happened which eventually makes this attestation invalid.

Reply:
An attestation must reference the previous crosslink. There are scenarios where we have already overwritten the previous crosslink and cannot validate the attestation to include it on chain. by adding the prev array, we ensure we have the data to validate attestations within the context of prev and current epoch